File: src/main/webapp/storage/src/app/dto/Record.ts
import {Item} from "./Item";

export class Record {
  private readonly _id: number;
  private readonly _count: number;
  private readonly _item: Item;


  constructor(id: number, count: number, item: Item) {
    this._id = id;
    this._count = count;
    this._item = item;
  }


  get id(): number {
    return this._id;
  }

  get count(): number {
    return this._count;
  }

  get item(): Item {
    return this._item;
  }
}

File: src/main/java/com/sonet/storage/model/record/Record.java
package com.sonet.storage.model.record;

import com.sonet.storage.model.item.Item;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Table(name = "record",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "item_id")
        })
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Record {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long count;

    @OneToOne
    private Item item;

    public Record(Long count, Item item) {
        this.count = count;
        this.item = item;
    }
}
